DigitalLearningSolutions.Web/ServiceFilter/ValidateAllowedDlsSubApplication.cs

```diff
@@ -1,26 +1,37 @@
-﻿namespace DigitalLearningSolutions.Web.Attributes
+﻿namespace DigitalLearningSolutions.Web.ServiceFilter
 {
-    using System;
     using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Web.Controllers;
+    using DigitalLearningSolutions.Web.Controllers.LearningSolutions;
     using DigitalLearningSolutions.Web.Helpers;
     using DigitalLearningSolutions.Web.Models.Enums;
     using Microsoft.AspNetCore.Mvc;
     using Microsoft.AspNetCore.Mvc.Filters;
     using Microsoft.AspNetCore.Mvc.ModelBinding;
+    using Microsoft.FeatureManagement;
 
-    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
-    public class ValidateAllowedDlsSubApplicationAttribute : Attribute, IActionFilter
+    public class ValidateAllowedDlsSubApplication : IActionFilter
     {
         private readonly string applicationArgumentName;
+        private readonly IFeatureManager featureManager;
+        private readonly IEnumerable<DlsSubApplication> validApplications;
 
-        public ValidateAllowedDlsSubApplicationAttribute(string applicationArgumentName = "dlsSubApplication")
+        public ValidateAllowedDlsSubApplication(
+            string[] validApplicationNames,
+            IFeatureManager featureManager,
+            string applicationArgumentName = "dlsSubApplication"

```

Comment:
As mentioned in the Knowledge Sharing Session, this can probably be replaced by using the same methodology as in SetDlsSubApplication.